sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -1267,6 +1394,8 @@ class HistGradientBoostingRegressor(RegressorMixin, BaseHistGradientBoosting):
           features.
         - str array-like: names of categorical features (assuming the training
           data has feature names).
+        - `"by_dtype"`: Pandas categorical dtypes are considered categorical.
```

Comment:
How about `"infer_by_dtype"`? It is a bit longer but a bit more explicit.
Just for discussion.

Reply:
I like this. Or maybe `"infer_from_dtype"`.

Reply:
In [`673c76d` (#26411)](https://github.com/scikit-learn/scikit-learn/pull/26411/commits/673c76dd95c4cb8a1955a0cfa3305009a6956885), I updated the keyword argument to `"from_dtype"`.